From fc6e2d9e6fa459f53c2b4a49478a6eb274ff5d79 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 28 May 2013 16:23:16 +0800
Subject: [PATCH 394/505] ENGR00230949 multi 8 instance dec test with some VPU
 error.100%

Bug: Instance quited because it failed to lock VPU mutex
[WARN] VPU mutex couldn't be locked before timeout expired or get lock
failure

Root cause:
Some instance(s) may not get a chance to be scheduled until timeout

Solution:
Yield schedule each frame explicitly. This may be kernel dependant
and may be removed on customer platform.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/dec.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 102fca4..586e835 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -1049,6 +1049,11 @@ decoder_start(struct decode *dec)
 
 		ret = vpu_DecGetOutputInfo(handle, &outinfo);
 
+		/* In 8 instances test, we found some instance(s) may not get a chance to be scheduled
+		 * until timeout, so we yield schedule each frame explicitly.
+		 * This may be kernel dependant and may be removed on customer platform */
+		usleep(0);
+
 		if ((dec->cmdl->format == STD_MJPG) &&
 		    (outinfo.indexFrameDisplay == 0)) {
 			outinfo.indexFrameDisplay = rotid;
-- 
1.8.0

